cmake/modules/ImportedTargetHelpers.cmake

```diff
@@ -1,6 +1,177 @@
+	if(NOT "${${_version_out}}" STREQUAL "" OR NOT TARGET "${_target}")
+		return()
+	endif()
+
+	# Return with a notice if cross-compiling, since we are unlikely to be able
+	# to run the compiled source
+	if(CMAKE_CROSSCOMPILING)
+		message(
+			"${_target} was found but the version could not be determined automatically.\n"
+			"Set the cache variable `${_version_out}` to the version you have installed."
+		)
+		return()
+	endif()
+
+	# Write the source code to a temporary file
+	string(SHA1 _source_hash "${_source}")
+	set(_source_file "${CMAKE_CURRENT_BINARY_DIR}/${_source_hash}.cpp")
+	file(WRITE "${_source_file}" "${_source}")
+
+	# Build and run the temporary file to get the version
+	# TODO CMake 3.25: Use the new signature for try_run which has a NO_CACHE
+	#                  option and doesn't require separate file management.
+	try_run(
+		_dvfs_run_result _dvfs_compile_result "${CMAKE_CURRENT_BINARY_DIR}"
+		SOURCES "${_source_file}"
+		LINK_LIBRARIES "${_target}"
+		CXX_STANDARD 17
+		RUN_OUTPUT_VARIABLE _run_output
+		COMPILE_OUTPUT_VARIABLE _compile_output
+	)
+
+	# Clean up the temporary file
+	file(REMOVE "${_source_file}")
+
+	# Set the version if the run was successful, using a cache variable since
+	# this version check may be relatively expensive. Otherwise, log the error
+	# and inform the user.
+	if(_dvfs_run_result EQUAL "0")
+		set("${_version_out}" "${_run_output}" CACHE INTERNAL "Version of ${_target}")
+	else()
+		message(DEBUG "${_compile_output}")
+		message(
+			"${_target} was found but the version could not be determined automatically.\n"
+			"Set the cache variable `${_version_out}` to the version you have installed."
+		)
+	endif()
+endfunction()
+
+# Search for a package using config mode. If this fails to find the desired
+# target, use the specified fallbacks and add the target if they succeed. Set
+# the variables `prefix_LIBRARY`, `prefix_INCLUDE_DIRS`, and `prefix_VERSION`
+# if found for the caller to pass to `find_package_handle_standard_args`.
+#
+# Usage:
+# 	_find_package_config_mode_with_fallback(
+#		<package_name>                 # The package to search for with config mode
+#		<target_name>                  # The target to expect from config mode, or define if not found
+# 		LIBRARY_NAMES names...         # Possible library names to search for as a fallback
+# 		INCLUDE_NAMES names...         # Possible header names to search for as a fallback
+# 		[PKG_CONFIG <pkg_config_name>] # The pkg-config name to search for as a fallback
+# 		[LIBRARY_HINTS hints...]       # Locations to look for libraries
+# 		[INCLUDE_HINTS hints...]       # Locations to look for headers
+#		[DEPENDS dependencies...]      # Dependencies of the target - added to INTERFACE_LINK_LIBRARIES, and will fail if not found
+#		[PREFIX <prefix>]              # The prefix for result variables - defaults to the package name
+# 	)
+function(_find_package_config_mode_with_fallback _fpcmwf_PACKAGE_NAME _fpcmwf_TARGET_NAME)
```

Comment:
I would recommend renaming this function to mimic the tool's proper name `pkg-config`.  I think this is a bit easier to read, given the abundance of the word "package" in helper scripts. 

```diff
- _find_package_config_mode_with_fallback
+ _find_pkg_config_mode_with_fallback
```

The variables may also benefit form a rename, but I couldn't think of a suggestion that was better than what is already provided.

Reply:
> `# The package to search for with config mode`

Ok, this can be disregarded, I realize that the function was named after [CMake's config mode](https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure) first, and `pkg-config` as the fallback.  Safe to resolve.